proxy/http2/HPACK.cc

```diff
@@ -946,22 +922,25 @@ hpack_encode_header_block(HpackIndexingTable &indexing_table, uint8_t *out_buf,
 
   MIMEFieldIter field_iter;
   for (MIMEField *field = hdr->iter_get_first(&field_iter); field != nullptr; field = hdr->iter_get_next(&field_iter)) {
```

Comment:
Why doesn't this use the STL paradigm of iterators with `begin` / `end`?

Reply:
Because `MIMEHdr` doesn't follow STL paradigm.

Reply:
That was actually my question - perhaps we should take `MIMEHdr` to do that.

Reply:
Filed an issue > https://github.com/apache/trafficserver/issues/7472